src/open_inwoner/configurations/bootstrap/zgw.py

```diff
@@ -0,0 +1,309 @@
+    """
+    Configure the required Service to establish a connection with the Formulieren API
+    """
+
+    verbose_name = "Formulieren APIs configuration"
+    required_settings = [
+        "FORMULIEREN_API_ROOT",
+        "FORMULIEREN_API_CLIENT_ID",
+        "FORMULIEREN_API_SECRET",
+    ]
+    enable_setting = "ZGW_API_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        return bool(Service.objects.filter(api_root=settings.FORMULIEREN_API_ROOT))
+
+    def configure(self):
+        organization = settings.OIP_ORGANIZATION or settings.ENVIRONMENT
+        org_label = f"Open Inwoner {organization}".strip()
+
+        form_service, created = Service.objects.update_or_create(
+            api_root=settings.FORMULIEREN_API_ROOT,
+            defaults={
+                "label": "Formulieren API",
+                "api_type": APITypes.orc,
+                "oas": settings.FORMULIEREN_API_ROOT,
+                "auth_type": AuthTypes.zgw,
+                "client_id": settings.FORMULIEREN_API_CLIENT_ID,
+                "secret": settings.FORMULIEREN_API_SECRET,
+                "user_id": settings.FORMULIEREN_API_CLIENT_ID,
+                "user_representation": org_label,
+            },
+        )
+        if not created:
+            form_service.oas = settings.FORMULIEREN_API_ROOT
+            form_service.client_id = settings.FORMULIEREN_API_CLIENT_ID
+            form_service.secret = settings.FORMULIEREN_API_SECRET
+            form_service.user_id = settings.FORMULIEREN_API_CLIENT_ID
+            form_service.user_representation = org_label
+            form_service.save()
+
+    def test_configuration(self):
+        """
+        actual testing is done in final step
+        """
+
+
+class ZGWAPIsConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the ZGW settings and set any feature flags or other options if specified
+    """
+
+    verbose_name = "ZGW APIs configuration"
+    enable_setting = "ZGW_API_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        zgw_config = OpenZaakConfig.get_solo()
+        return (
+            bool(zgw_config.zaak_service)
+            and bool(zgw_config.catalogi_service)
+            and bool(zgw_config.document_service)
+            and bool(zgw_config.form_service)
+        )
+
+    def configure(self):
+        config = OpenZaakConfig.get_solo()
+        config.zaak_service = Service.objects.get(api_root=settings.ZAKEN_API_ROOT)
+        config.catalogi_service = Service.objects.get(
+            api_root=settings.CATALOGI_API_ROOT
+        )
+        config.document_service = Service.objects.get(
+            api_root=settings.DOCUMENTEN_API_ROOT
+        )
+        config.form_service = Service.objects.get(
+            api_root=settings.FORMULIEREN_API_ROOT
+        )
+
+        # General config options
+        if settings.ZAAK_MAX_CONFIDENTIALITY:
+            config.zaak_max_confidentiality = settings.ZAAK_MAX_CONFIDENTIALITY
```

Comment:
Related to my comment above: here we set `config.zaak_max_confidentiality` to the value of `ZAAK_MAX_CONFIDENTIALITY`. But if the latter is `None`, we use the default defined by `OpenZaakConfig`. Why not use that default in our `settings/base.py`?